airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -232,6 +261,38 @@ def get_resource_type(self, resource_type, region_name=None, config=None):
             resource_type, endpoint_url=endpoint_url, config=config, verify=self.verify
         )
 
+    @cached_property
+    def conn(self):
+        """Get the underlying boto3 client (cached).
+
+        The return value from this method is cached for efficiency.
+
+        :return: boto3.client or boto3.resource for the current
+            client/resource type and region
+        :rtype: boto3.client() or boto3.resource()
+        :raises AirflowException: self.client_type or self.resource_type are not
+            populated. These are usually specified to this class, by a subclass
+            __init__ method.
+        """
+        if self.client_type:
+            return self.get_client_type(self.client_type, region_name=self.region_name)
+        elif self.resource_type:
+            return self.get_resource_type(self.resource_type, region_name=self.region_name)
+        else:
+            raise AirflowException(
+                'Either self.client_type or self.resource_type'
+                ' must be specified in the subclass')
```

Comment:
In the `__init__` you are validating and here you are only returning the correct boto client, right?
Also.. what do you think of a different exception type like `ValueError` for example?

What do you think of this?

```suggestion
        if self.client_type:
            return self.get_client_type(self.client_type, region_name=self.region_name)
        if self.resource_type:
            return self.get_resource_type(self.resource_type, region_name=self.region_name)
        raise ValueError('Could not get boto3 client!')
```

Reply:
Yes, in the `__init__` I do the actual validation. The only way the client_type/resource_type could change after that is if a subclass changes it directly. It would represent a bug in the subclass implementation. Based on the Python docs on Exceptions, a NotImplementedError sounds like the best fit.